ui/checkboxradio.js

```diff
@@ -0,0 +1,276 @@
+/*!
+ * jQuery UI Checkboxradio @VERSION
+ * http://jqueryui.com
+ *
+ * Copyright 2014 jQuery Foundation and other contributors
+ * Released under the MIT license.
+ * http://jquery.org/license
+ *
+ * http://api.jqueryui.com/checkboxradio/
+ */
+(function( factory ) {
+	if ( typeof define === "function" && define.amd ) {
+
+		// AMD. Register as an anonymous module.
+		define([
+			"jquery",
+			"./core",
+			"./widget"
+		], factory );
+	} else {
+
+		// Browser globals
+		factory( jQuery );
+	}
+}(function( $ ) {
+
+var formResetHandler = function() {
+		var form = $( this );
+		setTimeout(function() {
+			form.find( ".ui-checkboxradio" ).checkboxradio( "refresh" );
+		});
+	},
+	escapeId = new RegExp( /([!"#$%&'()*+,./:;<=>?@[\]^`{|}~])/g );
+
+$.widget( "ui.checkboxradio", {
+	version: "@VERSION",
+	options: {
+		disabled: null,
+		label: null,
+		icon: true,
+		classes: {
+			"ui-checkboxradio-label": "ui-corner-all",
+			"ui-checkboxradio-icon": "ui-corner-all"
+		}
+	},
+
+	_getCreateOptions: function() {
+		var disabled,
+			that = this,
+			options = this._super() || {};
+
+		this._readLabel();
+
+		this.originalLabel = "";
+		this.label.contents().not( this.element ).each( function() {
+			that.originalLabel += ( this.nodeType === 3 ) ? $( this ).text() : this.outerHTML;
+		});
+
+		disabled = this.element[ 0 ].disabled;
+		if ( disabled != null ) {
+			options.disabled = disabled;
+		}
+
+		if ( this.originalLabel ) {
+			options.label = this.originalLabel;
+		}
+
+		return options;
+	},
+
+	_create: function() {
+		var formElement = $( this.element[ 0 ].form );
+
+		// We don't use _on and _off here because we want all the checkboxes in the same form to use
+		// a single handler which handles all the checkboxradio widgets in the form
+		formElement.off( "reset" + this.eventNamespace, formResetHandler );
+		formElement.on( "reset" + this.eventNamespace, formResetHandler );
+
+		if ( this.options.disabled == null ) {
+			this.options.disabled = this.element[ 0 ].disabled || false;
+		}
+
+		this._readType();
+
+		this._enhance();
+
+		this._on({
+			"change": "_toggleClasses",
+			"focus": function() {
+				this._addClass( this.label, null, "ui-state-focus ui-visual-focus" );
+			},
+			"blur": function() {
+				this._removeClass( this.label, null, "ui-state-focus ui-visual-focus" );
+			}
+		});
+	},
+
+	_readType: function() {
+		this.type = this.element[ 0 ].type;
+		if ( !/radio|checkbox/.test( this.type ) ) {
+			$.error( "Can't create checkboxradio widget for type " + this.type );
+		}
+	},
+
+	_readLabel: function() {
+		var ancestor, labelSelector,
+			parent = this.element.closest( "label" );
```

Comment:
This only needs to run if `control.labels` is undefined or empty, could optimize that at the prize of a bit of complexity.